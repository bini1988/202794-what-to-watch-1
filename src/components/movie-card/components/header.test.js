import React from "react";
import renderer from "react-test-renderer";
import Header from "./header";
import card from "../../../mocks/movie-card";

it(`Header correctly renders default markup`, () => {
  const tree = renderer.create(
      <Header
        className="new-class"
        card={card}>
        {`Childs elements`}
      </Header>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
it(`Header correctly renders markup with component prop`, () => {
  const Component = (props) => (
    <div {...props}/>
  );
  const tree = renderer.create(
      <Header
        className="new-class"
        card={card}
        component={Component}>
        {`Childs elements`}
      </Header>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
