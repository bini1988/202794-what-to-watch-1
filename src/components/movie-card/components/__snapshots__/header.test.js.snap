// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header correctly renders default markup 1`] = `
<div
  className="movie-card__header new-class"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="War of the Worlds"
      src="https://es31-server.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg"
    />
  </div>
  Childs elements
</div>
`;

exports[`Header correctly renders markup with component prop 1`] = `
<div
  className="movie-card__head new-class"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="War of the Worlds"
      src="https://es31-server.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg"
    />
  </div>
  Childs elements
</div>
`;
